mage_integrations/mage_integrations/sources/stripe/__init__.py

```diff
@@ -1179,7 +1181,9 @@ def new_request(self, method, url, params=None, headers=None):
     # set the config and state in prior to check the authentication in the discovery mode itself.
     configure_stripe_client()
 
-    if source.discover_mode:
+    if source.should_test_connection:
+        pass
```

Comment:
I saw you implemented the test_connection method. why don't you call it here?

Reply:
When the main method for the stripe source is called, it tries to connect to the stripe client by default, so calling it again would be redundant.